### ISSUE TITLE
REQUEST- Engineering Presets

### ISSUE THREAD

**Issue by amtuttle** [daid/EmptyEpsilon#106]
It would be nice if there was a way to shift energy quickly between systems for engineering. It would be great to have presets of some sort.

**Comment by Fouindor** [comment]
Wouldn't that defeat a part of engineering purpose ? 
In one hand, it is always a very busy station and would relieve it a bit. On the other hand, isn't that part of the fun ?

I think it can be implemented without too much difficulty. How much presets would be available ?

**Comment by Wrongtown** [comment]
I think allowing binding of numeric keys allows a reasonable amount of pre-sets without making it too easy (and matches the feature-set available to Artemis SBS). As it is I think the need to select a system before making adjustments offers a greater challenge when in a tight spot if compared to the Artemis sliders.

**Suggestion**
Remove "0" from the custom pre-set pool and use it as a "systems nominal" key. This would split coolant equally across all systems and set all systems to default power, sort of an emergency override Engineering can use to establish a baseline before re-adjusting.

**Comment by daid** [comment]
The real question is where to put it on the UI? (With the background info that I reserved the top right corner of every crew station for stuff like main screen controls and possible scenario related controls later on)

**Comment by nallath** [comment]
Adding pre-sets might defeat the purpose of most of the skill challenge of the engineering imho. You would just pre-create a set of settings and you'd swich between those with a single key.

**Comment by kwadroke** [comment]
Going with the Artemis SBS example...
For a convention or similar setting with crews coming and going, presets really don't make sense.
For longer/multiple games with the same crew, presets can be a life saver. Some of the standard presets I've seen is: one for general cruising, one for low power, and another for battles. While it could reduce the skill some, it frees up some of their time to handle repairs and keep a closer eye on power consumption instead of being in panic mode all of the time. 
I do like to have my engineers running around like a chicken with their heads cut off :) When I captain the ship during conventions, I usually give engineering orders to give more power to different systems even if I really don't need it. 
For the LARP I'm planning, I want to give the engineering team more to do like calibrate the engines when they get out of alignment. I'll have to add some features to the code to allow this, but I don't want them to have to baby sit the controls just to get ready for battle. I want them to run around the room repairing things.

**Comment by kwadroke** [comment]
I've done some work on presets using the httpserver. It will require either running the game in windowed mode and running a browser, or running a second screen/computer with just the browser.
I'm still working on balancing coolant for a battle mode, else I'd upload it as a Gist.

**Comment by Wrongtown** [comment]
@daid you raise a good point about the UI, though there's a decent-sized area available directly under the Self-Destruct button... I don't think it would have to take up much space, I imagine you could have a Record State button with a Select State button, and only on selection of either of these do the numbers display for selection over the top of everything else like Communications in the Relay screen. 

Alternatively it *could* be entirely based on Keystrokes (and therefore unfortunately not available for Android engineering stations). For example SHIFT+1 to bind the current state to 1, so keypress of 1 tells the client to return the sliders to that state. That would allow for some pretty funky engineering stations with physical buttons for anyone looking to build them.

Personally I wouldn't think that allowing these suddenly makes Engineering too easy, they've still got plenty to keep them stressed with everyone crying for more power. This could be an option set when the server is stood up for those who don't like it, much like the Science Scanning settings.

**Comment by Jhovall** [comment]
I agree that the option should be there for advanced players, there is still a lot of small tinkering involved and it irritates me a bit that for every battle I have to go back to the same settings as engineer. The small tinkering is what makes the engineering interesting, and not the repeatable, chill-fight-or flight settings.

**Comment by hemmond** [comment]
I think the ENG presets could be interesting addition to the game - provided that it could be disabled from GM screen (in the same way "can selfdestruct"/"can Combat Maneuver" works. Feature available to some ships and not available to others. 

But presets can be implemented already through HTTP Server (and therefore easily run on wifi-enabled tablet/smartphone). Which solves the problem with not enough real estate on Engineering screen.
